src/uwtools/config/formats/ini.py

```diff
@@ -62,6 +63,7 @@ def dump_dict(path: OptionalPath, cfg: dict) -> None:
         # when an INI contains multiple sections. Unfortunately, it also adds a newline after the
         # _final_ section, resulting in an anomalous trailing newline. To avoid this, write first to
         # memory, then strip the trailing newline.
+        assert depth(cfg) <= INIConfig.DEPTH
```

Comment:
I used asserts here due `config_check_depths` requiring `(Config, str)`. I tried a variations where `config_check_depths` would handle Union[Config, dict] but went with asserts here rather than an if split in the check, although it would be a fair choice

Reply:
moved inside of config_check_depths for consistency